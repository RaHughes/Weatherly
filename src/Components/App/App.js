import React, { Component }from 'react';
import './App.scss';
import CurrentWeather from '../CurrentWeather/CurrentWeather';
import FiveDayForecast from '../FiveDayForecast/FiveDayForecast';
import Form from '../Form/Form';

class App extends Component {
  constructor() {
    super()
    this.state = { 
      city: '',
      country: '',
      data: []
     }
  }

  loadCity = (city, country) => {
    fetch(`http://api.openweathermap.org/data/2.5/forecast?q=${city},${country}?id=524901&APPID=f485a72a1e30b64d056573740d5064a3`)
    .then(data => data.json())
    .then(data => this.setState({data: data}))
    .catch(error => console.log(error))
  }

  findCity = (city, country) => {
    this.setState({ city: city, country: country})
    this.loadCity(city, country)
    console.log(this.state.city)
  }

  render() {
    return (
      <main className='App'>
        <h1 className='title'> Simple Weather </h1>
        {this.state.data.length === 0 ? 
        <header>
        <Form findCity={this.findCity}/>
        </header> : 
        <div className='main_section'>
        <button className='back_button' onClick={() => window.location.reload()}> Back </button>
        <CurrentWeather data={this.state.data}/>
        <h2 className='five_day_header'>Five Day Forecast</h2>
        <FiveDayForecast list={this.state.data.list}/>
        </div>}
      </main>
    )
  }
}


export default App;
