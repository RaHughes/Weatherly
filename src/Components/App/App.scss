* {
  margin: 0%;
  font-family: 'Bebas Neue', cursive;
  font-size: 25px;
}
.App {
  text-align: center;
  font-family: 'Bebas Neue', cursive;
  background-color: darkgray;

  h1 {
    background-color: lightgray;
    font-size: 50px;
    margin-bottom: 2%;
    border: 4px solid black;
    padding: 2%;
  }

  .main_section {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .back_button {
    height: 35px;
    width: 40%;
    cursor: pointer;
    &:hover {
      box-shadow: 0 0 10px blue;
      transition: 0.4s;
    }
  }

  .five_day_header {
    border-bottom: 4px solid black;
  }
}



