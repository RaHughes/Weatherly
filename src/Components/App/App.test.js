import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('Should match the snapshot of the DOM', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot();
  })

  it('Should be able to find a City', () => {
    const wrapper = shallow(<App />)
    const mockCity = {name: 'Denver'}
    const mockCountry = {country: 'usa'}
    const loadCity = jest.fn()

    wrapper.instance().findCity(mockCity, mockCountry)
    expect(wrapper.state('city')).toEqual(mockCity)
    expect(wrapper.state('country')).toEqual(mockCountry)
    // expect(loadCity).toHaveBeenCalledWith(mockCity, mockCountry)
  })
})
