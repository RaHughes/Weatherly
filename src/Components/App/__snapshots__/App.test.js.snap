// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match the snapshot of the DOM 1`] = `ShallowWrapper {}`;
